### PR TITLE
[BUU] Prepares specs for enabling admin_style_v3 by default

### PR DESCRIPTION
**Pull request by filipefurtad0**
This is already covered on the v3 version of the spec, under ./spec/system/admin/products_v3/products_spec.rb:452

#### What? Why?

- contributes to #12627

Addresses the bullet points:
- spec/system/admin/products_spec.rb:283
- spec/system/admin/subscriptions/smoke_tests_spec.rb:240
- improves specs as per [review comments](https://github.com/openfoodfoundation/openfoodnetwork/pull/11645#pullrequestreview-2152471490)

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/system/admin/subscriptions/smoke_tests_spec.rb [hunk 1]
```diff
@@ -242,15 +242,21 @@
...
+        visit admin_products_url
+
+        product_selector = row_containing_name(shop_product2.name)
+        delete_option_selector = "a[data-controller='modal-link'].delete"
+        delete_button_selector = "input[type=button][value='Delete product']"
+        modal_selector = "div[data-modal-target=modal]"
+
+        within product_selector do
+          page.find(".vertical-ellipsis-menu").click
+          page.find(delete_option_selector).click
```
Comment (dacook) [inline spec/system/admin/subscriptions/smoke_tests_spec.rb:254]: Would this work?
```suggestion
          click_link "Delete"
```

File: spec/system/admin/subscriptions/smoke_tests_spec.rb [hunk 2]
```diff
@@ -242,15 +242,21 @@
...
+        end
+        within modal_selector do
+          page.find(delete_button_selector).click
```
Comment (dacook) [inline spec/system/admin/subscriptions/smoke_tests_spec.rb:257]: And `click_button`?

**Review by dacook — COMMENTED** [review]
Thanks for moving on this one quickly! I have an idea to simplify the code:

File: spec/system/admin/subscriptions/smoke_tests_spec.rb (hunk 1 above)
Comment (filipefurtad0) [reply spec/system/admin/subscriptions/smoke_tests_spec.rb:254]: Thanks for the early review @dacook :pray: 

Hum.. Unfortunately not. I think first, we'd need to open the actions sub menu, which we could do with:

`page.find(".vertical-ellipsis-menu").click`

But even after that, it does not seem to work. This is the element we're trying to click I think, which is neither a button nor a link:

```
<a class="delete" data-action="click->modal-link#setModalDataSetOnConfirm click->modal-link#open" data-controller="modal-link" data-modal-link-modal-dataset-value="{&quot;data-delete-path&quot;:&quot;/admin/products_v3/11&quot;}" data-modal-link-target-value="product-delete-modal">
Delete
</a>
```

I'm not sure how we could simplify that spec line...

File: spec/system/admin/subscriptions/smoke_tests_spec.rb (hunk 2 above)
Comment (filipefurtad0) [reply spec/system/admin/subscriptions/smoke_tests_spec.rb:257]: The confirm/cancel modal has is indeed elements which are `type=button`:

`<input class="button icon-plus red" data-action="click->products#delete_product click->modal#close" id="modal-confirm-button" type="button" value="Delete product" data-delete-path="/admin/products_v3/22">`

So, `click_button "Delete product"` works :smile:

Comment (filipefurtad0) [reply spec/system/admin/subscriptions/smoke_tests_spec.rb]: @dacook I went ahead and applied your proposed improvement on the the [v3 products spec](https://github.com/openfoodfoundation/openfoodnetwork/pull/12652/commits/4e5ab0f77fd92510b34bf8062de929db569e9169).

**Comment by filipefurtad0** [comment]
I think to fully address issue https://github.com/openfoodfoundation/openfoodnetwork/issues/12627 we'd need to include a migration (something [like this](https://github.com/openfoodfoundation/openfoodnetwork/pull/10693/files)?), to enable `admin_style_v3` by default. But I don't think we're there just yet - we're looking at the [16th of July](https://community.openfoodnetwork.org/t/backoffice-ui-product-list-uplift-rollout/2921/35?u=filipefurtado) - so maybe this PR is ready for review for now.

File: spec/system/admin/subscriptions/smoke_tests_spec.rb (hunk 1 above)
Comment (dacook) [reply spec/system/admin/subscriptions/smoke_tests_spec.rb:254]: Oh, fair enough. This is a classic example where the spec reveals that the markup needs improving!

I think it _should_ be a link. Hrm, but that's complicated, we can't make a useful link without JS, so it's a JS-only element. So... let's just leave this as-is, thanks for figuring out this one.

File: spec/system/admin/order_cycles/list_spec.rb [hunk 3]
```diff
@@ -73,7 +81,9 @@
 
     # I can load more order_cycles
     expect(page).not_to have_selector "#listing_order_cycles tr.order-cycle-#{oc7.id}"
-    trigger_click(:button, "Show 30 more days")
+    # binding.pry
+    # trigger_click(:button, "Show 30 more days")
```
Comment (dacook) [inline spec/system/admin/order_cycles/list_spec.rb:85]: I think these comments can be removed ;)

```suggestion
```

**Review by dacook — APPROVED** [review]
Great improvements, well done!

**Comment by dacook** [comment]
Rebased on master (git kindly resolved conflicts automatically for me 🤩 )

```
Applying: Simplifies specs, by replacing selector declaration with click_button
Using index info to reconstruct a base tree...
A	spec/system/admin/products_v3/products_spec.rb
Falling back to patching base and 3-way merge...
Auto-merging spec/system/admin/products_v3/actions_spec.rb
```

**Review by mkllnk — APPROVED** [review]
Great teamwork! The failing specs seem unrelated.

**Pull request merged**